include/seqan3/core/simd/view_iota_simd.hpp

```diff
@@ -0,0 +1,248 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::counted_simd_iterator and seqan3::views::iota_simd.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/std/ranges>
+
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/core/simd/concept.hpp>
+#include <seqan3/core/simd/simd_algorithm.hpp>
+#include <seqan3/core/simd/simd_traits.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Implements a special version of a counted iterator over a simd vector.
+ * \ingroup simd
+ * \implements std::forward_iterator
+ *
+ * \tparam index_t The type of the index; must model seqan3::simd::simd_concept.
+ *
+ * \details
+ *
+ * Uses a simd count vector to increment the counted iterator. This seems to be in general faster than calling
+ * seqan3::simd::fill when dereferencing the iterator, although this is just a constant and fast operation.
+ */
+template <simd_concept index_t>
+class counted_simd_iterator
+{
+private:
+    //!\brief The currently represented count.
+    index_t count_simd{};
+    //!\brief The count in scalar representation.
+    size_t count_scalar{};
```

Comment:
Is this really faster? I always thought you can access `count_simd[0]` without cost.

Reply:
I think it was the same speed but given that we also need to access it for computinf the size etc. But I meassure it again just to be certain.